misk-hibernate/src/main/kotlin/misk/hibernate/Transacter.kt

```diff
@@ -49,6 +49,12 @@ interface Transacter {
    */
   fun <T> replicaRead(block: (session: Session) -> T): T
 
+  /**
+   * Runs a read on master first then tries it on replicas on failure. This method is here only for
+   * health check purpose for standby regions.
+   */
+  fun <T> failSafeRead(block: (session: Session) -> T): T
```

Comment:
I think this can be made an extension method as well right? You should be able to build it purely on top of the existing Session interface.